v0/destinations/hs/util.js

```diff
@@ -17,7 +133,168 @@ const getEmailAndUpdatedProps = properties => {
   };
 };
 
+/* NEW API util functions */
+
+/**
+ * look for the contact in hubspot and extract its contactId for updation
+ * @param {*} destination
+ * @returns
+ */
+const searchContacts = async (message, destination, lookupField = null) => {
+  const { Config } = destination;
+  let res;
+  let contactId = null;
+  const traits = getFieldValueFromMessage(message, "traits");
+  let propertyName;
+
+  if (lookupField) {
+    propertyName = lookupField;
+  } else {
+    propertyName = traits[`${Config.lookupField}`];
+  }
+
+  const value = traits[`${propertyName}`];
+
+  if (!value) {
+    throw new CustomError(
+      `[HS] Identify:: '${Config.lookupField}' key name i.e provided in config is required under traits for contact lookup`,
+      400
+    );
+  }
+
+  const requestData = {
+    filterGroups: [
+      {
+        filters: [
+          {
+            propertyName,
+            value,
+            operator: "EQ"
+          }
+        ]
+      }
+    ],
+    sorts: ["ascending"],
+    properties: [propertyName],
+    limit: 2,
+    after: 0
+  };
+
+  if (Config.authorizationType === "newPrivateAppApi") {
```

Comment:
authorisation part for choosing API key or accessToken seems redundant, as it being used in different other functions too. Lets create a util function and encapsulate it

Reply:
It actually varies from endpoint to endpoint and different message types and also while searching for contacts. cannot make it common for all.